### PR TITLE
gravel/tests: add missing type hints

### PR DESCRIPTION
**Pull request by jecluis**
Signed-off-by: Joao Eduardo Luis \<joao@suse.com>

### CODE REVIEW

File: src/gravel/tests/conftest.py [hunk 1]
```diff
@@ -85,7 +91,9 @@ def ceph_conf_file_fs(
 
 
 @pytest.fixture()
-def get_data_contents(fs: fake_filesystem.FakeFilesystem):
+def get_data_contents(  # type: ignore
+    fs: fake_filesystem.FakeFilesystem
+) -> str:
```
Comment (mgfritch) [inline src/gravel/tests/conftest.py:96]:
```suggestion
def get_data_contents(
    fs: fake_filesystem.FakeFilesystem
) -> Generator[Callable[[str, str], str], None, None]:
```

these ones are fun, but I think the above is correct

File: src/gravel/tests/conftest.py [hunk 2]
```diff
@@ -85,7 +91,9 @@ def ceph_conf_file_fs(
...
     def _get_data_contents(dir: str, fn: str):
```
Comment (mgfritch) [inline src/gravel/tests/conftest.py:97]:
```suggestion
    def _get_data_contents(dir: str, fn: str) -> str:
```

File: src/gravel/tests/conftest.py [hunk 3]
```diff
@@ -99,10 +107,10 @@ def _get_data_contents(dir: str, fn: str):
         with open(os.path.join(dir, fn), 'r') as f:
             contents = f.read()
         return contents
-    yield _get_data_contents
+    yield _get_data_contents  # type: ignore
```
Comment (mgfritch) [inline src/gravel/tests/conftest.py:110]:
```suggestion
    yield _get_data_contents
```

**Review by mgfritch — APPROVED** [review]
an improvement overall, but it would be nice to avoid adding more `type: ignore` ..

File: src/gravel/tests/conftest.py (hunk 1 above)
Comment (jhesketh) [reply src/gravel/tests/conftest.py:96]: I think you're correct, but eww :-p

**Review by jhesketh — APPROVED** [review]
lgtm! Thanks for that.

(Not sure if you want to fix up Mike's suggestion first or not)

File: src/gravel/tests/conftest.py (hunk 1 above)
Comment (jecluis) [reply src/gravel/tests/conftest.py:96]: tried that. It works to shut up mypy, but won't shut up pyright. That's because when it's called, it's not called as a generator, butt as a function (because pytest is doing magic in the background). So pyright will be annoyed because we're calling on a generator without actually using it as a generator.

TL;DR: pytest typing support sucks, and this was the closest I could get to make that function work.

File: src/gravel/tests/conftest.py (hunk 3 above)
Comment (jecluis) [reply src/gravel/tests/conftest.py:110]: This works if we're using the generator return, but not in any other way. While we get correctness for this function, we lose correctness (at least from the expected behavior point of view) for the rest of the tests that are using this function. We can't be telling the type checker we're yielding a function here and expect `get_data_contents` to be returning a string without calling on `next()`.

**Comment by jecluis** [comment]
Thanks. I'm kicking Mike's comments down the road to a time when someone (either me or whomever) has time to figure out how to type check pytest generators correctly. I actually started my endeavor from where Mike's comments are, and after an afternoon of insanity I ended up choosing to ignore types for that function to make the rest work.
